src/charm.py

```diff
@@ -86,6 +87,32 @@ class COSProxyCharm(CharmBase):
 
     _stored = StoredState()
 
+    mandatory_relation_pairs = {
```

Comment:
What is the intent of an item in `mandatory_relation_pairs`?  is it:
1. `{"name of mandatory relation": ["additional relations that we require one of", "if we have this relation"]}`
2. {"name of optional relation": ["list of relations that we require one of", "if we have this relation"]}`
?

based on [this](https://github.com/canonical/cos-proxy-operator/pull/145/files#diff-b9ed39bbc9c0387bd3e07da31d13373745534a1cd723d3e292c73496b12e307cR656) I think the intent is definition (1), but I don't think that solves #144.  #144 is a bug where cos-proxy is requiring the user to establish a `dashboards` relation even though they don't have/want one.  

I think here we're combining two concepts:
* a relation that the charm always requires (mandatory)
* a relation that the charm requires iff the charm has some other relation (eg: having one of `cos-agent` or `downstream-grafana-dashboard` if you have a `dashboards` relation)

where iiuc these should be separate.